Add Databot section to Positron docs (#163)
posit-dev/positron-docs: databot.qmd

The [Databot extension](https://open-vsx.org/extension/posit/databot) is for use with [Positron](https://positron.posit.co/). It uses functionality available only in Positron, so it will not work with VS Code or other Code OSS-based IDEs.

## Installation

Currently, there are a few steps to getting Databot up and running.

First, make sure you have a recent version of Positron (2025.08.0 is the minimum version).

<!-- vale Posit.Headings = NO -->

### Install the Databot extension

In Positron, choose the [Extensions view](extensions.qmd) from the Activity Bar on the left or use the keyboard shortcut {{< kbd mac=Command-Shift-X win=Ctrl-Shift-X linux=Ctrl-Shift-X >}} and search for “Databot”. Click the Install button to install the extension.

### Enable Databot

Databot is an experimental product currently in research preview and is not ready for production use. To enable Databot, you must first acknowledge this.

* In Positron, open the Command Palette {{< kbd mac=Command-Shift-P win=Ctrl-Shift-P linux=Ctrl-Shift-P >}} and run `Open User Settings`.  

> * In the search bar, type “Databot”.  

Author: Done!

> * You will see **Databot: Research Preview Acknowledgment**. In the text box, type “Acknowledged”.

Reviewer: These are not regular quotes right now, so we do need to change it.

> * You will see **Databot: Research Preview Acknowledgment**. In the text box, type “Acknowledged”.

Author: Changed a few, thanks!

### Start Databot

> In Positron, open the Command Palette {{< kbd mac=Command-Shift-P win=Ctrl-Shift-P linux=Ctrl-Shift-P >}} and run `Open Databot`.

Reviewer: ```suggestion
In Positron, open the Command Palette {{< kbd mac=Command-Shift-P win=Ctrl-Shift-P linux=Ctrl-Shift-P >}} and run _Open Databot_.
```

### Enable Positron Assistant 

> If [Positron Assistant](assistant.qmd) is not yet enabled, you will see the following message in the Databot panel: “To use Databot, enable Positron Assistant and Anthropic provider.”

Reviewer: ```suggestion
If [Positron Assistant](assistant.qmd) is not yet enabled, you will see the following message in the Databot panel: "To use Databot, enable Positron Assistant and Anthropic provider."
```

Click on the button labeled **Help me set it up**. This will open a dialog box.

In section 2 of the dialog box, **Enable Positron Assistant**, if it says “Positron Assistant is not yet enabled” and there are any red check boxes, then click on the button labeled **Update settings**. The check boxes will turn green.

### Enable Anthropic as a model provider

:::{.callout-note}

> ## Prerequisites: 

Reviewer: ```suggestion
```

For consistency on the site as a whole

For this step, you will need an Anthropic API key.

:::

In section 3 of the dialog box, **Add Anthropic provider**, click on the button labeled **Click here**. This will open a dialog box to configure Language Model Providers.

* Select **Anthropic** and enter your API key.  

Close the dialog boxes. Databot is now ready to use!

## Using Databot

Databot will now ask you what you want to do, and will guide you through a data analysis process.

To start Databot in another Positron session, open the Command Palette {{< kbd mac=Command-Shift-P win=Ctrl-Shift-P linux=Ctrl-Shift-P >}} and run `Open Databot`.

> For more information about Databot, please review the [Databot introduction blogpost](https://posit.co/blog/introducing-databot/).

Reviewer: ```suggestion
For more information about Databot, please review the [Databot introduction blog post](https://posit.co/blog/introducing-databot/).
```

## FAQ

### Who is Databot for?

* Databot is currently intended for experienced R and Python users who are familiar with data analysis.

### Are you sure it’s not for people who aren’t comfortable writing their own code? Why not?

> * As far as we can tell, today’s models cannot be trusted to do open-ended data exploration without at least occasionally making mistakes. That’s why Databot’s user experience is intentionally code-forward, and a Databot analysis is not truly complete until the user has carefully reviewed important subsets of the model-generated code. [Learn more here](https://posit.co/blog/databot-is-not-a-flotation-device).

Reviewer: ```suggestion
* As far as we can tell, today’s models cannot be trusted to do open-ended data exploration without at least occasionally making mistakes. That’s why Databot’s user experience is intentionally code-forward, and a Databot analysis is not truly complete until the user has carefully reviewed important subsets of the model-generated code. Learn more at [Databot is not a flotation device](https://posit.co/blog/databot-is-not-a-flotation-device).
```

### How is Databot different from Positron Assistant?

> * [Positron Assistant](https://positron.posit.co/assistant.html) is an _AI coding assistant_: it helps you write code or modify code on disk and can execute arbitrary code in Agent mode. Databot is a purpose-built _exploratory data analysis_ agent. It writes and executes its own code on the fly to help you understand your data. It doesn’t care very much about the code you’ve already written.

Reviewer: ```suggestion
* [Positron Assistant](assistant.qmd) is an _AI coding assistant_; it helps you write code or modify code on disk and can execute arbitrary code in Agent mode. Databot is a purpose-built _exploratory data analysis_ agent. It writes and executes its own code on the fly to help you understand your data. It doesn’t care very much about the code you’ve already written.
```

### What data sources can Databot use? Will it work with databases? Parquet files on S3?

* In theory, anything you can reasonably do from an R or Python console is possible in Databot, as long as Claude knows how to do it. For accessing data that isn’t on disk, you may need to provide a little more guidance to Databot to get started, like giving it a snippet of code that shows how you normally connect to your data source in Python.

### What LLMs can Databot use?

* Databot is currently calibrated to work with Claude Sonnet 4. It also works well with Claude 3.5 Sonnet v2. Claude Opus performs similarly to Sonnet 4 but is more expensive, so we recommend Sonnet 4 in most cases. We do not recommend Claude 3.7 Sonnet because it tends to do far too much work before returning control to the user.

### Are OpenAI, Gemini, and other models supported?

* Not at this time, though we hope to add support in the future once they are exposed in Positron Assistant.

> <!-- vale Posit.Headings = YES -->

Reviewer: ```suggestion
```

Remove this and add "Databot" to the allowlist instead.
